autoload/codi.vim

```diff
@@ -186,6 +186,34 @@ function! s:user_au(au)
   endif
 endfunction
 
+" If percent / float width, calculate based on buf width
+" If result leaves less than |minwidth| / 20 columns of working space,
+" return a width that leaves |minwidth| / 20 columns to write code in
+" Else, return absolute width as given
+function! s:pane_width()
+  let width = s:get_opt('width')
+  let width = type(width) == type('') ? width : string(width)
+
+
+  if match(width, '[.%]') > -1
+    let raw          = str2float(substitute(width, '%', '', 'g'))
+    let clamped      = raw > 100.0 ? 100.0 : (raw < 0.0 ? 0.0 : raw)
+    let min_width    = exists('&winwidth') ? &winwidth : 20
+    let buffer_width = winwidth(bufwinnr('%'))
+    let result       = ceil((buffer_width / 100.0) * clamped)
+
+    if (buffer_width - result) < min_width
```

Comment:
Can be simplified as `return max(min_width, min(buffer_width - min_width, result))`.

Reply:
See previous `max` comment. P.S. your arguments are wrong according to my `:h max`:

![image](https://user-images.githubusercontent.com/3225058/44516760-2e943380-a6c6-11e8-9b2f-44ac0d572d21.png)

Reply:
If I understand you correctly, you want to turn all the floats into integers using `ceil` so that they can be used in `max` when wrapped in a list. This is something I thought of too but wasn't sure if the lost precision was worth it but if it is cool with you it's cool with me. It will make the code a bit cleaner and easier to understand as well 👍 

I am already working on the next commit, all that remains is this part so I'll be back soon. (My internet failed twice while trying to write this comment :/)